### PR TITLE
#57 Guided Tour

### PR DESCRIPTION
**Pull request by kahboom**
Not perfect; some bugs, but it works!

### CODE REVIEW

**Comment by kahboom** [comment]
screenshot 2017-10-31 11 10 24

Needs styling and additional steps.

File: src/app/app.component.ts [hunk 1]
```diff
@@ -67,21 +68,15 @@ export class AppComponent implements OnInit, AfterViewInit {
...
+              private nav: NavigationService,
+              private modalService: ModalService,
+              public tourService: TourService) {}
 
   ngOnInit() {
-    this.productBuild = this.config.getSettings(
-      'branding',
-      'productBuild',
-      false,
-    );
```
Comment (gashcrumb) [inline src/app/app.component.ts]: erm, why are these lines being removed?

**Comment by gashcrumb** [comment]
Some lint errors:

```
src/app/connections/create-page/create-page.component.ts[30, 46]: Unnecessary trailing comma
src/app/connections/create-page/create-page.component.ts[30, 47]: missing whitespace
```

File: src/app/app.component.ts (hunk 1 above)
Comment (kahboom) [reply src/app/app.component.ts]: This must have been removed when resolving a merge conflict; I'll add them again now, sorry!

**Comment by kahboom** [comment]
Thanks!

**Comment by kahboom** [comment]
Changes:
- Dismissible tour service that is self-contained per component, with some caveats that need to be sorted out (more on that below).
- Styling to PatternFly was also completed.
- Steps that can be demoed were added, otherwise the steps that would be in the way and inhibit development were not added (more on that below as well).

Some screenshots:

screenshot 2017-11-02 22 54 58

screenshot 2017-11-02 22 55 14

screenshot 2017-11-02 22 55 54


Still outstanding:
- A bug with not being able to dismiss the tour for steps that are anchored in lazy-loaded module component templates, nor being able to properly go back/forward. It's glitchy and sometimes the Next button shows for no reason for those steps.
- Missing several steps, such as for creating an integration. These take a few minutes to add, but to get it working properly so that you can dismiss them or navigate through the tour is a different thing altogether. Need to get that first issue resolved and these steps can be added in the next sprint. Same goes for the dat amapper.

**Pull request merged**